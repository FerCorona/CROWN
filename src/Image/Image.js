import React from 'react';
import { IMAGE_SIZE } from '../Common/Constants';

import image from '../Common/Assets/crown.png';

require('./styles.scss');

const Image = ({ src, size, shadow }) => {
  const style = {
    width: `${IMAGE_SIZE[size]}`,
    height: `${IMAGE_SIZE[size]}`
  };
  if (shadow) {
    style.boxShadow = '0 4px 8px 0 rgba(0,0,0,0.2)';
  }
  return (
    <div className='CROWNImage' style={style}>
      <img src={src} style={style} />
    </div>
  );
};

Image.defaultProps = {
  src: image,
  size: 'medium',
  shadow: true
};

export default Image;