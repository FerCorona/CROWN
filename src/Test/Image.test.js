import React from 'react';
import renderer from 'react-test-renderer';
import Image from '../Image/index';

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <Image />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});