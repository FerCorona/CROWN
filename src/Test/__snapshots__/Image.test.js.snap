// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="CROWNImage Shadow "
  style={
    Object {
      "height": "350px",
      "width": "300px",
    }
  }
>
  <div
    className="ImageBlur"
    style={
      Object {
        "backgroundImage": "linear-gradient(45deg, var(--red-1), var(--pink-1))",
        "backgroundPosition": "center",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "cover",
        "position": "relative",
      }
    }
  >
    <div
      className="Blur"
    >
      <div
        className="Image"
        style={
          Object {
            "backgroundImage": "linear-gradient(45deg, var(--red-1), var(--pink-1))",
            "backgroundPosition": "center",
            "backgroundRepeat": "no-repeat",
            "backgroundSize": "cover",
            "position": "relative",
          }
        }
      />
    </div>
  </div>
</div>
`;
